### PR TITLE
Releaseherald/docs

### PR DESCRIPTION
**Pull request by kn-ms**
## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [ ] News fragment present for release notes
- [ ] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: releaseherald/docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+**_Default:_** `"(?P<version>(\d*)\.(\d*)\.(\d*))"`
+
+### `news_fragments_directory`
+
+: The path to the news fragment directory, if relative, it is relative to the repository root.  
+**_Default:_** `news_fragment`
+
+### `news_file`
+
+: Path of the base news file. This is the file that will be extended wit the news for versions that `releaseherald`
```
Comment (Pyifan) [inline releaseherald/docs/configuration.md:30]: s/wit/with

File: releaseherald/docs/index.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+`releaseherald` is a Testplan internal utility that can generate release notes for Testplan. The only help it needs from
+developers is to collect meaningful news fragments in a directory, and apply consistent labels.
+
+> -- Wait a bit I have already heard this somewhere.  
+> -- Yes we were really inspired by [towncrier](https://github.com/twisted/towncrier), but we had a bit different opinions
+
+## Philosophy
+
+We very much agree [towncrier's philosophy](https://github.com/twisted/towncrier#Philosophy):
```
Comment (Pyifan) [inline releaseherald/docs/index.md:11]: agree with?

File: releaseherald/docs/index.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+- version control in git
+- the release notes for a version is fabricated from the news fragments created/modified between that two release
+- the releases number can be worked out from git labels
+
+## Basics
+
+`releaseherald` is a command line tool it needs python 3.7+ to run. It needs a config file in the root of the git repo
+it should be called `releaseherald.toml`. If it is used in a python project it can read it's configuration
+from `pyproject.toml` from `[tool.releaseherald]`. It is possible to start without any configuration, then one need to
```
Comment (Pyifan) [inline releaseherald/docs/index.md:26]: s/need/needs

File: releaseherald/docs/index.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+1.0.0 (2021-10-15)
+------------------
+
+* 1, before first release
+* 2, before first release
+```
+
+This tool try not to have many opinions, so all the above defaults can be tweaked through configuration or command line.
+New fragments not need to be removed between releases, no need to commit the generated release notes either,
```
Comment (Pyifan) [inline releaseherald/docs/index.md:57]: do not need

File: releaseherald/docs/version_template.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Version template
+
+!!! warning "This is not yet written"
```
Comment (Pyifan) [inline releaseherald/docs/version_template.md:3]: Eh, is this expected?

Comment (kn-ms) [reply releaseherald/docs/version_template.md:3]: I would like the placeholder page there, as todo item, the best to write when I will have metadata extraction as the structure the templates works on will be much closer to final by that. Putting it into a warning box make it stands out showing that we are working on this.

**Review by Pyifan — APPROVED** [review]

**Pull request merged**